src/core/plugins/index.js

```diff
@@ -78,11 +79,12 @@ class PluginIndex {
   action(action) {
```

Comment:
This could be converted to an `async` function, avoiding at least `Promise.resolve` and then/catch if you don't mind a try...catch block.

Reply:
while that would be the better style and we should move to it, i would rather keep the structure similar to the other functions for now without re-writing too much.